client/src/components/TextInput.jsx

```diff
@@ -11,10 +11,20 @@ const TextInput = ({
   maxLength,
   multiline,
   disabled,
+  variant,
 }) => {
   const classes = `${label && "text-input__label"}
     ${subLabel ? "text-input__label--with-sublabel" : ""}`;
 
+  const handleChange = (e) => {
+    if (variant === "number") {
+      e.target.value = e.target.value.replace(/\D/, "");
+    }
+
+    onChange(e);
+  };
+  const inputType = variant === "number" ? "numeric" : "text";
```

Comment:
i'll approve. But can you change this to `inputMode`? Seems better to just keep same naming convention as html.

Reply:
i like saving as a variable here tho